docs/src/main/tut/TypedML.md

```diff
@@ -15,17 +29,226 @@ spark.sparkContext.setLogLevel("WARN")
 
 import spark.implicits._
 ```
+
+## Example 1: predict a continuous value using a `TypedRandomForestRegressor`
+
+In this example, we want to predict `field3` of type Double from `field1` and `field2` using 
+a `TypedRandomForestRegressor`.
+
+### Training
+
+As with Spark ML API, we use a `TypedVectorAssembler` (the type-safe equivalent of `VectorAssembler`) to compute feature vectors:
+
+```tut:silent
+import frameless._
+import frameless.syntax._
+import frameless.ml._
+import frameless.ml.feature._
+import frameless.ml.regression._
+import org.apache.spark.ml.linalg.Vector
+```
+
+```tut:book
+case class Data(field1: Double, field2: Int, field3: Double)
+
+val trainingData = TypedDataset.create(
+  Seq.fill(10)(Data(0D, 10, 0D))
+)
+
+case class Features(field1: Double, field2: Int)
+val assembler = TypedVectorAssembler[Features]
+
+case class DataWithFeatures(field1: Double, field2: Int, field3: Double, features: Vector)
+val trainingDataWithFeatures = assembler.transform(trainingData).as[DataWithFeatures]
+```
+
+Then, we train the model:
+
+```tut:book
+case class RFInputs(field3: Double, features: Vector)
+val rf = TypedRandomForestRegressor[RFInputs]
+
+val model = rf.fit(trainingDataWithFeatures).run()
+```
+
+As an example of added type-safety, `TypedRandomForestRegressor[RFInputs]` will compile only if the given `RFInputs` case class 
+contains only one field of type Double (the label) and one field of type Vector (the features):
+
+```tut:silent
+case class WrongRFInputs(labelOfWrongType: String, features: Vector)
+```
+
+```tut:book:fail
+TypedRandomForestRegressor[WrongRFInputs]
+```
+
+The subsequent `rf.fit(trainingDataWithFeatures)` call will compile only if `trainingDataWithFeatures` contains the same fields 
+(names and types) as RFInputs.
+
+```tut:book
+val wrongTrainingDataWithFeatures = TypedDataset.create(Seq(Data(0D, 1, 0D))) // features are missing
+```
+
+```tut:book:fail
+rf.fit(wrongTrainingDataWithFeatures) 
+```
+
+For new-comers to frameless, please note that `typedDataset.as[...]` is a type-safe cast, 
```

Comment:
I would remove the "For new-comers to frameless," and place this section where we first call `.as[]`